src/promptflow-core/promptflow/contracts/flow.py

```diff
@@ -1019,6 +1026,48 @@ def deserialize(data: dict) -> "FlexFlow":
             environment_variables=data.get("environment_variables") or {},
         )
 
+    @classmethod
+    def _from_dict(cls, flow_dag: dict, working_dir: Path, name=None) -> "FlexFlow":
+        """Load flow from dict."""
+        from promptflow._core.entry_meta_generator import generate_flow_meta
+
+        from .._utils.flow_utils import resolve_python_entry_file
+
+        Flow._update_working_dir(working_dir)
+        if name is None:
+            name = flow_dag.get("name", _sanitize_python_variable_name(working_dir.stem))
+        flow_dag["name"] = name
+
+        entry = flow_dag.get("entry")
+        entry_file = resolve_python_entry_file(entry=entry, working_dir=working_dir)
+
+        meta_dict = generate_flow_meta(
+            flow_directory=working_dir,
+            source_path=entry_file,
+            data=flow_dag,
+        )
+        return cls.deserialize(meta_dict)
+
+    def get_connection_names(self, environment_variables_overrides: Dict[str, str] = None):
+        """Return connection names."""
+        from promptflow.core._model_configuration import MODEL_CONFIG_NAME_2_CLASS
+
+        connection_names = super().get_connection_names(environment_variables_overrides=environment_variables_overrides)
+        try:
+            init_params = os.environ.get(PROMPTFLOW_FLOW_INIT_CONFIG, "{}")
+            init_dict: dict = json.loads(init_params)
```

Comment:
We can't assume that this environment variable is set when user run `pf flow build` locally, right? cc @D-W- @wangchao1230 

With current logic, we will always dump no connections?